libintervalxt/src/intervalxt/detail/interval_exchange_transformation.ipp

```diff
@@ -186,13 +186,66 @@ class IntervalExchangeTransformation<Length>::Implementation {
     return top.begin()->label.length() == bottom.begin()->label.length();
   }
 
-  // Return whether there is a periodic trajectory via Boshernitzan's
-  // algorithm.
+  // Test whether the Boshernitzan criterion guarantees minimality. If
+  // true then it is certainly minimal. If false it needs not be that there
+  // is a saddle connection.
   bool boshernitzanMinimal(void) const {
-    return false;
+      return false;
```

Comment:
```
if constexpr (std::is_same_v<Length, mpz_class> || std::is_same_v<Length, mpq_class>) {
  return false;
} else {
  ...;
}
```

Reply:
```
template <typename L>
bool boshernitzanMinimal(const IntervalExchangeTransformation<L>& iet) {
    throw std::logic_error("not implemented");
}
```

```
template <>
bool boshernitzanMinimal<mpz_class>(…){…}
```

Reply:
```
template <typename Length>
class Boshernitzan {
    constexpr bool is_minimal = false;
};

template <>
class Boshernitzan<mpz_class> {
    constexpr bool is_minimal = true;
};
```

Reply:
This does not work since the template parameter is `Length` that wraps the actual `int`, `long`, `mpz_class`, `mpq_class`, `renf_elem_class`, ...